File: java/src/main/java/com/dokmatiq/docgen/model/ImageData.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/** Embedded image data for a document bookmark. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ImageData(
        String bookmarkName,
        String base64,
        Integer width,
        Integer height,
        String altText
) {}

File: php/src/Model/ImageData.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

use Dokmatiq\DocGen\Internal\Serializer;

/** Embedded image data for a document bookmark. */
final class ImageData
{
    public function __construct(
        public readonly string $bookmarkName,
        public readonly string $base64,
        public readonly ?int $width = null,
        public readonly ?int $height = null,
        public readonly ?string $altText = null,
    ) {}

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return Serializer::toArray($this);
    }
}
